src/main/java/ee/tuleva/onboarding/payment/PaymentController.java

```diff
@@ -38,11 +42,32 @@ public PaymentLink getPaymentLink(
   @Operation(summary = "Redirects user to payment success")
   public RedirectView getPaymentSuccessRedirect(
       @RequestParam("payment_token") String serializedToken) {
-    Optional<Payment> payment = paymentService.processToken(serializedToken);
-    if (payment.isPresent()) {
-      return new RedirectView(frontendUrl + "/3rd-pillar-success");
+    Optional<Payment> paymentOptional = paymentService.processToken(serializedToken);
+
+    return paymentOptional
```

Comment:
We should keep this kind of business logic out of the controller.

Reply:
It's basically just routing based on Payment

Reply:
TODO: User redirect URL to avoid if statements in this controller